lib/util/git/index.ts

```diff
@@ -902,12 +903,78 @@ export function getUrl({
   });
 }
 
-export async function pushCommitAsRef(
+const remoteRenovateRefs = new Set<string>();
+
+/**
+ *
+ * Non-branch refs allow us to store git objects without triggering CI pipelines.
+ * It's useful for API-based branch rebasing.
+ *
+ * @see https://stackoverflow.com/questions/63866947/pushing-git-non-branch-references-to-a-remote/63868286
+ *
+ */
+export async function pushCommitToRenovateRef(
   commitSha: string,
   refName: string
 ): Promise<void> {
-  await git.raw(['update-ref', refName, commitSha]);
-  await git.raw(['push', '--force', 'origin', refName]);
+  const fullRefName = `refs/renovate/${refName}`;
+  await git.raw(['update-ref', fullRefName, commitSha]);
+  await git.raw(['push', '--force', 'origin', fullRefName]);
+  remoteRenovateRefs.add(fullRefName);
+}
+
+/**
+ *
+ * Removes all remote "refs/renovate/*" refs in two steps:
+ *
+ * Step 1: list refs
+ *
+ *   $ git ls-remote origin "refs/renovate/*"
+ *
+ *   > cca38e9ea6d10946bdb2d0ca5a52c205783897aa        refs/renovate/foo
+ *   > 29ac154936c880068994e17eb7f12da7fdca70e5        refs/renovate/bar
+ *   > 3fafaddc339894b6d4f97595940fd91af71d0355        refs/renovate/baz
+ *   > ...
+ *
+ * Step 2:
+ *
+ *   $ git push --delete origin refs/renovate/foo refs/renovate/bar refs/renovate/baz
+ *
+ */
+export async function clearRenovateRefs(): Promise<void> {
+  if (gitInitialized && remoteRenovateRefs.size > 0) {
+    logger.debug(`Clear Renovate refs: refs/renovate/*`);
+    try {
+      const repoCache = getCache();
+      const fetchSkipsTotal = 15;
+      repoCache.renovateRefsFetchSkipsCounter ??= fetchSkipsTotal;
+      if (repoCache.renovateRefsFetchSkipsCounter > 0) {
+        repoCache.renovateRefsFetchSkipsCounter -= 1;
+      } else {
+        repoCache.renovateRefsFetchSkipsCounter = fetchSkipsTotal;
+
+        const rawOutput = await git.raw([
+          'ls-remote',
+          config.url,
+          'refs/renovate/*',
+        ]);
+
+        rawOutput
+          .split(newlineRegex)
+          .map((line) => line.replace(regEx(/[0-9a-f]+\s+/i), ''))
+          .filter(is.truthy)
+          .forEach((ref) => {
+            remoteRenovateRefs.add(ref);
+          });
+      }
+
+      const clearCmd = ['push', '--delete', 'origin', ...remoteRenovateRefs];
+      remoteRenovateRefs.clear();
+      await git.raw(clearCmd);
+    } catch (err) /* istanbul ignore next */ {
+      logger.debug({ err }, `Clear Renovate refs: error`);
```

Comment:
Would it be worth storing the output of the `git.raw` command in case it's useful to log here?

Reply:
I think failed command outputs are included in `err` object, aren't they?